add support for multiple channels, aggregates into 1 database

tiw48 opened the pull request.

Commit:
add support for multiple channels, aggregates into 1 database

Commit:
update bot to accommodate 2 people

Commit:
Update README.md

Commit:
Change bot Message Format a little

michaelmdresser commented on build_msg_db.py:
```diff
@@ -4,24 +4,15 @@
-# from a single channel, though this could
-# easily be extended
-general_channel_id = int(sys.argv[1])
+# channel IDs to read messages from, separated by spaces
```
Remove this line.

michaelmdresser commented on build_msg_db.py:
```diff
@@ -33,25 +24,33 @@ async def build_db_for_channel(channel_id):
 # https://stackoverflow.com/questions/63846749/how-to-send-message-without-command-or-event-discord-py
 # once the task is done, you can ctrl-c this process
-client.loop.create_task(build_db_for_channel(general_channel_id))
+client.loop.create_task(build_db_for_channel(sys.argv))
```
Maybe assert `len(sys.argv) > 1`?

michaelmdresser commented on build_msg_db.py:
```diff
@@ -33,25 +24,33 @@ async def build_db_for_channel(channel_id):
+        print("getting channel ", channel_id)
+        channel = client.get_channel(channel_id)
+        if channel is None:
+            raise Exception("channel is none")
```
Now that we're looping, maybe log an error with the channel ID and keep going.

michaelmdresser commented on run_bot.py:
```diff
@@ -9,6 +9,22 @@
 # the channel id to restrict sending messages to
 botspam_channel_id = int(sys.argv[1])
 
+def generate_message(serialized_user):
```
I think it would be better if this just took a user ID and handled all of the markov chain loading and generating logic, include file path.

michaelmdresser commented on run_bot.py:
```diff
@@ -20,37 +36,44 @@ async def on_message(message):
-    return
-
+        output_message = generate_message(serialized_target)
+    else:
```
Instead of this, use `filter` or something similar to make a list of mentions without the bot, then loop over that list.

michaelmdresser commented on run_bot.py:
```diff
@@ -20,37 +36,44 @@ async def on_message(message):
+        else:
+            target = message.mentions[0]
+            secondary = message.mentions[2]
+        # if you would like mentions instead of nicknames, uncomment these two lines and comment the two after them.
+        # targetID = '<@'+str(target.id)+'>'
+        # secondaryID = '<@'+str(secondary.id)+'>'
+        targetID = target.nick
+        secondaryID = secondary.nick
```
nice

tiw48 replied:
fixed bug for users without a nickname

Commit:
Fixed bug for users without a nickname

Commit:
made fixes based on pull review

build_msg_db.py
- change error logic in case of invalid channel ID
- added check for CLI length
run_bot.py
- changed generate_message to take a target.id
- changed 3 mention logic to use filter
- changed output_message according to generate_message changes

michaelmdresser commented:
Test comment, please ignore.
